src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -27,17 +27,42 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.server")
-
-	registerPeriod = 5 * time.Minute
+	log               = golog.LoggerFor("flashlight.server")
+	registerPeriod    = 5 * time.Minute
+	frontingProviders = map[string]func(*http.Request) bool{
+		// WARNING: If you add a provider here, keep in mind that Go's http
+		// library normalizes all header names so the first letter of every
+		// dash-separated "word" is uppercase while all others are lowercase.
+		// Also, try and check more than one header to lean on the safe side.
+		"cloudflare": func(req *http.Request) bool {
+			return hasHeader(req, "Cf-Connecting-Ip") || hasHeader(req, "Cf-Ipcountry") || hasHeader(req, "Cf-Ray") || hasHeader(req, "Cf-Visitor")
+		},
+		"cloudfront": func(req *http.Request) bool {
+			return hasHeader(req, "X-Amz-Cf-Id") || headerMatches(req, "User-Agent", "Amazon Cloudfront")
+		},
+	}
 )
 
+func headerMatches(req *http.Request, name string, value string) bool {
+	for _, entry := range req.Header[name] {
+		if entry == value {
+			return true
+		}
+	}
+	return false
+}
+
+func hasHeader(req *http.Request, name string) bool {
+	return req.Header[name] != nil
+}
```

Comment:
We don't have that no. Fine to leave for now, but we should probably start building something like that up sometime soon.